src/libraries/System.Linq/tests/ElementAtOrDefaultTests.cs

```diff
@@ -11,73 +11,345 @@ public class ElementAtOrDefaultTests : EnumerableTests
         [Fact]
         public void SameResultsRepeatCallsIntQuery()
         {
-            var q = from x in new[] { 9999, 0, 888, -1, 66, -777, 1, 2, -12345 }
-                    where x > int.MinValue
-                    select x;
-
-            Assert.Equal(q.ElementAtOrDefault(3), q.ElementAtOrDefault(3));
+            var q = Repeat(_ => from x in new[] { 9999, 0, 888, -1, 66, -777, 1, 2, -12345 }
```

Comment:
Curious what the motivation for this method is. It took me a while to grasp but it seems to be creating 3 identical IEnumerable instances? Why would that be needed?

Reply:
The original unit test creates a source, then calls ElementAtOrDefault twice on the same source, and asserts the result.

I am keeping the consistency.

Reply:
Sure, but for all intents and purposes the source enumerable in the test is immutable. We should not expect noticeable difference no matter how many times it has been enumerated. I would recommend keeping the tests as simple as possible (in this case and most other cases using the `Repeat` method, `q` can be enumerated as many times as needed. Assuming this were not the case, perhaps `MemberData` is preferable?